care/abdm/api/viewsets/abha_number.py

```diff
@@ -0,0 +1,56 @@
+from django.db.models import Q
+from django.http import Http404
+from rest_framework.decorators import action
+from rest_framework.mixins import RetrieveModelMixin
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from care.abdm.api.serializers.abha_number import AbhaNumberSerializer
+from care.abdm.models import AbhaNumber
+from care.abdm.utils.api_call import HealthIdGateway
+from care.utils.queryset.patient import get_patient_queryset
+
+
+class AbhaNumberViewSet(
+    GenericViewSet,
+    RetrieveModelMixin,
+):
+    serializer_class = AbhaNumberSerializer
+    model = AbhaNumber
+    queryset = AbhaNumber.objects.all()
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
```

Comment:
Can we rewrite the method to get the object first and then check for permission, The current query might be expensive to execute.